### PR TITLE
Add Spearmanr Metric Card

### PR DESCRIPTION
**Pull request by emibaylor**

### CODE REVIEW

File: metrics/spearmanr/spearmanr.py [hunk 1]
```diff
@@ -38,19 +37,39 @@
 
 _KWARGS_DESCRIPTION = """
 Args:
-    predictions: Predicted labels, as returned by a model.
-    references: Ground truth labels.
+    - **`predictions`** (`list` of `float`): Predicted labels, as returned by a model.
+    - **`references`** (`list` of `float`): Ground truth labels.
+    - **`return_pvalue`** (`bool`): If `True`, returns the p-value. If `False`, returns
+                                    only the spearmanr score. Defaults to `False`.
```
Comment (lhoestq) [inline metrics/spearmanr/spearmanr.py:43]: You can use the same format as other metric docstrings:
```suggestion
    - predictions (`List[float]`): Predicted labels, as returned by a model.
    - references (`List[float]`): Ground truth labels.
    - return_pvalue (`bool`): If `True`, returns the p-value. If `False`, returns
            only the spearmanr score. Defaults to `False`.
```

File: metrics/spearmanr/spearmanr.py [hunk 2]
```diff
@@ -38,19 +37,39 @@
...
 Returns:
-    spearmanr: Spearman correlation coefficient.
+    -  **`spearmanr`** (`float`): Spearman correlation coefficient.
+    - **`p-value`** (`float`): p-value. **Note**: is only returned
+                        if `return_pvalue=True` is input.
```
Comment (lhoestq) [inline metrics/spearmanr/spearmanr.py:47]:
```suggestion
    - spearmanr (`float`): Spearman correlation coefficient.
    - p-value (`float`): p-value. **Note**: is only returned if `return_pvalue=True` is input.
```

File: metrics/spearmanr/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+The same example, but that also returns the pvalue:
+```python
+>>> spearmanr_metric = datasets.load_metric("spearmanr")
+>>> results = spearmanr_metric.compute(references=[1, 2, 3, 4, 5], predictions=[10, 9, 2.5, 6, 4], return_pvalue=True)
+>>> print(results)
+{'spearmanr': SpearmanrResult(correlation=-0.7, pvalue=0.1881204043741873)}
+>>> print(results['spearmanr'].correlation)
+-0.7
+>>> print(round(results['spearmanr'].pvalue, 2))
+0.19
```
Comment (lhoestq) [inline metrics/spearmanr/README.md:77]: (nit) I would have expected it to return a "pvalue" field with the actual value inside it. Is there a reason to keep everything ?

**Review by lhoestq — COMMENTED** [review]
Thank you ! A few comments:

File: metrics/spearmanr/README.md (hunk 3 above)
Comment (sashavor) [reply metrics/spearmanr/README.md:77]: I think keeping both values is useful, depending on what people use this for!

**Comment by sashavor** [comment]
changes made! @lhoestq let me know what you think

File: metrics/spearmanr/spearmanr.py [hunk 4]
```diff
@@ -38,19 +37,38 @@
...
-    predictions: Predicted labels, as returned by a model.
-    references: Ground truth labels.
+    - predictions (`List[float]`): Predicted labels, as returned by a model.
+    - references (`List[float]`): Ground truth labels.
+    - return_pvalue (`bool`): If `True`, returns the p-value. If `False`, returns
+            only the spearmanr score. Defaults to `False`.
 Returns:
-    spearmanr: Spearman correlation coefficient.
+    - spearmanr (`float`): Spearman correlation coefficient.
+    - p-value (`float`): p-value. **Note**: is only returned if `return_pvalue=True` is input.
```
Comment (lhoestq) [inline metrics/spearmanr/spearmanr.py:46]:
```suggestion
    spearmanr (`float`): Spearman correlation coefficient.
    p-value (`float`): p-value. **Note**: is only returned if `return_pvalue=True` is input.
```

File: metrics/spearmanr/spearmanr.py [hunk 5]
```diff
@@ -38,19 +37,38 @@
...
-    >>> spearmanr_metric = datasets.load_metric("spearmanr")
-    >>> results = spearmanr_metric.compute(references=[1, 2, 3, 4, 5], predictions=[10, 9, 2.5, 6, 4])
-    >>> print(results)
-    {'spearmanr': -0.7}
+    Example 2:
+        >>> spearmanr_metric = datasets.load_metric("spearmanr")
+        >>> results = spearmanr_metric.compute(references=[1, 2, 3, 4, 5],
+        ...                                     predictions=[10, 9, 2.5, 6, 4],
+        ...                                     return_pvalue=True)
+        >>> print(results['spearmanr`])
```
Comment (lhoestq) [inline metrics/spearmanr/spearmanr.py:59]:
```suggestion
        >>> print(results['spearmanr'])
```

**Review by lhoestq — APPROVED** [review]
Thanks ! Re-formatted some docstrings and fixed the doctest (a backtick was used instead of a quote, it should fix the CI)

**Comment by lhoestq** [comment]
The CI fail is unrelated to this PR and fixed on master, feel free to merge :)

**Pull request merged**